tools/flambda_backend_objinfo.ml

```diff
@@ -160,41 +166,45 @@ let print_generic_fns gfns =
 
 
 let print_cmx_infos (ui, crc) =
-  print_general_infos Compilation_unit.output ui.ui_unit crc ui.ui_defines
-    (fun f -> List.iter f ui.ui_imports_cmi)
-    (fun f -> List.iter f ui.ui_imports_cmx);
-  begin match ui.ui_export_info with
-  | Clambda approx ->
+  print_general_infos Compilation_unit.output ui.uir_unit crc ui.uir_defines
+    (fun f -> List.iter f ui.uir_imports_cmi)
+    (fun f -> List.iter f ui.uir_imports_cmx);
+  begin match ui.uir_export_info with
+  | Clambda_raw approx ->
     if not !no_approx then begin
       printf "Clambda approximation:\n";
       Format.fprintf Format.std_formatter "  %a@." Printclambda.approx approx
     end else
       Format.printf "Clambda unit@.";
-  | Flambda1 export ->
+  | Flambda1_raw export ->
     if not !no_approx || not !no_code then
       printf "Flambda export information:\n"
     else
       printf "Flambda unit\n";
     if not !no_approx then begin
-      Compilation_unit.set_current (Some ui.ui_unit);
-      let root_symbols = List.map Symbol.for_compilation_unit ui.ui_defines in
+      Compilation_unit.set_current (Some ui.uir_unit);
+      let root_symbols = List.map Symbol.for_compilation_unit ui.uir_defines in
       Format.printf "approximations@ %a@.@."
         Export_info.print_approx (export, root_symbols)
     end;
     if not !no_code then
       Format.printf "functions@ %a@.@."
         Export_info.print_functions export
-  | Flambda2 None ->
+  | Flambda2_raw None ->
     printf "Flambda 2 unit (with no export information)\n"
-  | Flambda2 (Some cmx) ->
+  | Flambda2_raw (Some cmx) ->
     printf "Flambda 2 export information:\n";
     flush stdout;
+    let cmx =
+      Flambda2_cmx.Flambda_cmx_format.from_raw cmx
+        ~sections:Flambda_backend_utils.File_sections.empty
```

Comment:
I think it would be better to actually read the sections here (the cost is almost zero anyway), in case the printing code uses them at some point. (They might even be currently read, I'm not sure).